client.go

```diff
@@ -318,3 +308,25 @@ func (c ByteCredentials) getTransportCredentials() (credentials.TransportCredent
 	})
 	return creds, nil
 }
+
+func isValid(endpoint string) error {
+	host, port, err := net.SplitHostPort(endpoint)
+	if err != nil {
+		//nolint dunno howto convert this to errors.As
+		if addrErr, ok := err.(*net.AddrError); ok {
+			return fmt.Errorf("%s", addrErr.Err)
+		}
+		// shouldn't happen, but...
```

Comment:
I think this extra error handling can be skipped

Reply:
Done